### PR TITLE
Source bundle registration

### PR DESCRIPTION
**Pull request by Xiaochao8**

### CODE REVIEW

File: src/main/java/com/vmware/vipclient/i18n/VIPCfg.java [hunk 1]
```diff
@@ -37,6 +46,10 @@
 	// cache mode
 	private CacheMode cacheMode = CacheMode.MEMORY;
 
+	public final static String COMPONENT = "component";
+	public final static String RESOURCE = "resource";
```
Comment (linr211) [inline src/main/java/com/vmware/vipclient/i18n/VIPCfg.java]: both constants can be move to com.vmware.vipclient.i18n.util.ConstantsKeys class

File: src/main/java/com/vmware/vipclient/i18n/VIPCfg.java [hunk 2]
```diff
@@ -71,56 +86,33 @@ public static synchronized VIPCfg getInstance() {
...
-					.getString("cleanCache"));
-		if (prop.containsKey("machineTranslation"))
-			this.machineTranslation = Boolean.parseBoolean(prop
-					.getString("machineTranslation"));
-		if (prop.containsKey("i18nScope"))
-			this.i18nScope = prop.getString("i18nScope");
-		if (prop.containsKey("cacheExpiredTime"))
-			this.cacheExpiredTime = Long.parseLong(prop
-					.getString("cacheExpiredTime"));
+		StringService.loadResources(this.resources);
```
Comment (linr211) [inline src/main/java/com/vmware/vipclient/i18n/VIPCfg.java]: The VIPCfg class should provide a method getResources for the instance of StringService to call, so 'StringService.loadResources(this.resources)' should be removed?

File: src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java [hunk 3]
```diff
@@ -253,9 +252,14 @@ public String getString2(final String component,
 		String message = "";
 		String source;
 		try {
-			ResourceBundle rb = ResourceBundle.getBundle(bundle,
-					LocaleUtility.defaultLocale);
-			source = rb.getString(key);
+			MessagesDTO dto = new MessagesDTO();
```
Comment (linr211) [inline src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java]: How about this solution to load the bundle file to get source?
1 Remove 'bundle' parameter
2 Get the bundle value by component from the yaml config
3 Replace as:
 ResourceBundle rb = ResourceBundle.getBundle(**new_bundle**,
					LocaleUtility.defaultLocale)

File: src/main/java/com/vmware/vipclient/i18n/messages/service/StringService.java [hunk 4]
```diff
@@ -110,4 +115,44 @@ public boolean isStringAvailable() {
...
+	// Get source
+	public String getSource() {
+		Map<?, ?> messages = StringService.defaultResources.get(dto.getCompositStrAsCacheKey());
+		String result = (null == messages ? "" : (String) messages.get(this.dto.getKey()));
+		return null == result ? "" : result; 
+	}
+	
+	// load sources from local files.
+	@SuppressWarnings("unchecked")
+	public static void loadResources(List<Map<String, Object>> resources) throws IOException {
```
Comment (linr211) [inline src/main/java/com/vmware/vipclient/i18n/messages/service/StringService.java]: move this method to util?

File: src/main/java/com/vmware/vipclient/i18n/messages/service/StringService.java [hunk 5]
```diff
@@ -110,4 +115,44 @@ public boolean isStringAvailable() {
...
+		if (null == resources || resources.isEmpty()) {
+			throw new VIPJavaClientException("No resources are provided in config file!");
+		}
+		
+		for( Map<String, Object> e : resources) {
+			String comp = (String) e.get(VIPCfg.COMPONENT);
+			String res = (String) e.get(VIPCfg.RESOURCE);
+			
+			Map<Object, Object> messages;
+			if (res.endsWith(".json")) {
```
Comment (huihuiw01) [inline src/main/java/com/vmware/vipclient/i18n/messages/service/StringService.java]: For source file format, do we plan to support both json and properties?

File: src/main/java/com/vmware/vipclient/i18n/VIPCfg.java [hunk 6]
```diff
@@ -71,56 +86,33 @@ public static synchronized VIPCfg getInstance() {
...
 	}
-	
-	/**
-	 * initialize the instance by parameter
-	 * 
-	 * @param vipServer
-	 * @param productName
-	 * @param version
-	 */
-	public void initialize(String vipServer, String productName, String version) {
```
[the remark this answers is not in the document]
Comment (Xiaochao8) [reply src/main/java/com/vmware/vipclient/i18n/VIPCfg.java]: fixed

File: src/main/java/com/vmware/vipclient/i18n/VIPCfg.java (hunk 1 above)
Comment (Xiaochao8) [reply src/main/java/com/vmware/vipclient/i18n/VIPCfg.java]: fixed

File: src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java (hunk 3 above)
Comment (Xiaochao8) [reply src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java]: The bundle files may be json format, can't load them with this method.

File: src/main/java/com/vmware/vipclient/i18n/messages/service/StringService.java (hunk 4 above)
Comment (Xiaochao8) [reply src/main/java/com/vmware/vipclient/i18n/messages/service/StringService.java]: Moved to LocalSourceOpt

**Pull request closed without merging**